File: java-maven/src/main/java/com/example/AdminService.java
package com.example;

/**
 * AdminService — NOT_REACHABLE (Type A).
 *
 * This class has @Service annotation, so Spring will instantiate it
 * during component scan.  However, no @Controller or other bean
 * injects it via @Autowired, so its methods are never called from
 * any HTTP endpoint.
 *
 * CWE-78 (command injection) — NOT_REACHABLE: service never injected.
 * SECRET — NOT_REACHABLE: key defined but inaccessible.
 */
import org.springframework.stereotype.Service;
import java.io.IOException;

@Service
public class AdminService {

    // SECRET: Hardcoded admin token (NOT_REACHABLE — service never injected)
    private static final String ADMIN_TOKEN = "adm_live_spring_7mXq2K";

    /**
     * CWE-78 — NOT_REACHABLE (Type A): @Service exists but no controller
     * injects AdminService.
     */
    public String executeCommand(String cmd) throws IOException {
        // CWE-78: command injection — NOT_REACHABLE (Type A)
        Process p = Runtime.getRuntime().exec(cmd);
        return new String(p.getInputStream().readAllBytes());
    }

    /**
     * SECRET — NOT_REACHABLE (Type A): endpoint inaccessible.
     */
    public String getAdminToken() {
        return ADMIN_TOKEN;  // SECRET NOT_REACHABLE (Type A)
    }
}

File: kotlin-app/src/main/kotlin/com/example/AdminService.kt
package com.example

/**
 * AdminService — NOT_REACHABLE (Type A).
 *
 * This class has @Service annotation, so Spring will instantiate it.
 * However, no @Controller or other bean injects it via constructor
 * or @Autowired, so its methods are never called from HTTP endpoints.
 *
 * CWE-78 (command injection) — NOT_REACHABLE: service never injected.
 * SECRET — NOT_REACHABLE: key defined but inaccessible.
 */
import org.springframework.stereotype.Service

@Service
class AdminService {

    // SECRET: Hardcoded admin token (NOT_REACHABLE — service never injected)
    private val adminToken = "adm_live_kotlin_8kZp3Q"

    /** CWE-78 — NOT_REACHABLE (Type A): @Service exists but no controller injects it. */
    fun executeCommand(cmd: String): String {
        // CWE-78: command injection — NOT_REACHABLE (Type A)
        val process = Runtime.getRuntime().exec(arrayOf("sh", "-c", cmd))
        return process.inputStream.bufferedReader().readText()
    }

    /** SECRET — NOT_REACHABLE (Type A). */
    fun getAdminToken(): String = adminToken  // SECRET NOT_REACHABLE (Type A)
}
